### DISCUSSION TITLE
关于特效模式下的任务栏

### DISCUSSION THREAD

**Discussion by dheart-joe** [linuxdeepin/developer-center#5161] [Developer Discussion | 开发者讨论]
1、首先特效模式下的任务栏跟底下的边缘空隙都可以开一部车了，巨丑，说了也不改，无法理解，本来桌面就没多大的空间，还那么大的空隙，无语。
2、还有就是能不能有个全屏模式，像macos一样，放大就是全屏，不然像下面这样，应用放大后左右两边的空隙，真的是放个地球了。无法忍受
![image](https://github.com/linuxdeepin/developer-center/assets/10397085/c48b2b80-d62d-4696-99d9-95e76d4a6981)

**Comment by BLumia** [comment]
觉得下面占地方，考虑开智能隐藏或始终隐藏吧？下边的边框就是设计如此的，切到高效模式就没有悬浮的空隙了。

至于丑不丑，可能不同的人观点差距还是很大的，见过说这个很好看的，也见过说这个很丑的...

**Reply by dheart-joe** [reply]
你这句话就等于你不喜欢就不用的意思

**Reply by BLumia** [reply]
看上去你错误的理解了我的意思：我的意思是美丑是见仁见智的而不是说有人说丑就是丑，并提供了一些关于你提到的一些问题的现有解决方案。

**Comment by Decodetalkers** [comment]
任务栏的行为和wm是绑定的。而且任务栏有重构的计划，到时候再处理吧

**Reply by BLumia** [reply]
重构仍然会基于现有的设计人员提供的设计图和产品文档来进行实现，如果是希望修改现有的效果的话，其实应当尝试说服设计人员（（

**Reply by dheart-joe** [reply]
设计人员怎么联系

**Reply by BLumia** [reply]
可能去论坛反馈更方便一些？目前设计人员并不使用 GitHub，只有参与开发与测试的人员/贡献者用。

不过 @meiyixiang 产品这边也可以看看 :)

**Comment by GershonWang** [comment]
看到这个discussions，我也想讨论一句，其实这个特效样式很不错。只是针对不同分辨率的屏幕，这个dock栏上下边距真的无力吐槽。同样的dock栏在高分屏下的边距效果和在普清屏下的效果视觉美感真的差距挺大。这个问题等同于之前标题栏过宽的问题，高分屏下展示很合适尺寸，在普通分辨率屏下展示就很宽。

**Comment by meiyixiang** [comment]
这个问题设计定义的是10个PX，基于最基础的考虑是1920X1080的屏幕。

**Reply by GershonWang** [reply]
不能这么锁死尺寸大小。毕竟现在笔记本1366*768的分辨率还很普遍而且不在少数，这个尺寸应该针对不同的分辨率弹性适应

**Comment by meiyixiang** [comment]
待任务栏重构的时候看如何优化一下。

**Comment by pmzgit** [comment]
自从v20 23高效改成这个德行以后，就从来没用过这个特效模式
你们官方人员如果自认为这有别于win11或macos，比占用屏幕空间重要，那让这么多提出问题的人无话可说
但是你们如果觉得这个美？那你们在论坛以及github，开源中国等社区，发个投票看看

我记得两年前就有很多人提出这个浪费屏幕空间的问题了，就没别的方案，用户自定义难死了？

可别提设计人员了，一个操作系统的ui，交互，渲染风格，图标，每一个都很重要，这是deepin绝对竞争力之一.但你看看v20，不能说不行吧，但应用和设置的图标，得有40%，难看的要命

到23了交互才上来，渲染也提升了，图标主题有的还不错， 但是图标还不够精致，所以说别提你们的设计人员了，学学v15吧

**Reply by BLumia** [reply]
还是确实应当说服设计的。我们其实有计划从研发的角度允许定制成用户希望的布局，但无论我们的研发架构设计上如何灵活，默认行为一定会是设计和产品定义的...比如产品设计定义上是不允许自定义布局，那么我们架构上即便允许，除非设计与产品同意，否则不能直接提供自定义的选项出来...

**Comment by meiyixiang** [comment]
可以讨论下当前说到的，除了任务栏特效模式上下预留的空白空间大。任务栏时间和插件分为2行不合理还有哪些问题？可以发散讨论一下。集中整理意见整体优化。即使不能在核心功能中支持也可以考虑在定制工具中支持。

**Reply by dheart-joe** [reply]
任务栏特效模式下，建议有应用全屏的开关
     1、IDE：
                  像我们做开发的，就喜欢IDE全屏，不受其他信息干扰，而且屏幕利用率也很好，代码浏览也舒服。
     2、视频：
                  看视频的场景也是希望全屏，不希望底下还有一个大大的尾巴(工具栏-dock)，除了预留的空白空间大，而且这个尾巴左右两边还当飞机场。
